numalogic/udfs/trainer/_rds.py

```diff
@@ -0,0 +1,272 @@
+        except AttributeError as err:
+            raise ConfigNotFoundError("RDS config not found!") from err
+
+    def register_rds_fetcher_conf(
+        self, config_id: str, pipeline_id: str, conf: RDSFetcherConf
+    ) -> None:
+        """
+        Register RDSFetcherConf with the UDF.
+
+        Args:
+            config_id: Config ID
+            conf: RDSFetcherConf object
+        """
+        fetcher_id = f"{config_id}-{pipeline_id}"
+        self.pl_conf.rds_conf.id_fetcher[fetcher_id] = conf
+
+    def get_rds_fetcher_conf(self, config_id: str, pipeline_id: str) -> RDSFetcherConf:
+        """
+        Get RDSFetcherConf with the given ID.
+
+        Args:
+            config_id: Config ID
+
+        Returns
+        -------
+            RDSFetcherConf object
+
+        Raises
+        ------
+            ConfigNotFoundError: If config with the given ID is not found
+        """
+        fetcher_id = f"{config_id}-{pipeline_id}"
+        try:
+            return self.pl_conf.rds_conf.id_fetcher[fetcher_id]
+        except KeyError as err:
+            raise ConfigNotFoundError(
+                f"Config with ID {fetcher_id} not found in rds_conf!"
+            ) from err
+
+    def fetch_data(self, payload: TrainerPayload) -> Optional[pd.DataFrame]:
+        """
+        Fetch data from RDS.
+
+        Args:
+            payload: TrainerPayload object
+
+        Returns
+        -------
+            Dataframe
+        """
+        _start_time = time.perf_counter()
+
+        # TODO : Make this code generic and add to trainer utils
+        _config_id = payload.config_id
+        _pipeline_id = payload.pipeline_id
+        _metric_label_values = (
+            payload.composite_keys,
+            ":".join(payload.composite_keys),
+            _config_id,
+            _pipeline_id,
+        )
+        _stream_conf = self.get_stream_conf(_config_id)
+        _conf = _stream_conf.ml_pipelines[_pipeline_id]
+        _fetcher_conf = self.dataconn_conf.fetcher or (
+            self.get_rds_fetcher_conf(config_id=_config_id, pipeline_id=_pipeline_id)
+            if self.dataconn_conf.id_fetcher
+            else None
+        )
+        if not _fetcher_conf:
+            raise ConfigNotFoundError(
+                f"RDS fetcher config not found for config_id: {_config_id},"
+                f" pipeline_id: {_pipeline_id}!"
+            )
+
+        try:
+            query = build_query(
+                datasource=_fetcher_conf.datasource,
+                dimensions=_fetcher_conf.dimensions,
+                metrics=_fetcher_conf.metrics,
+                datetime_column_name=_fetcher_conf.datetime_column_name,
+                hash_query_type=_fetcher_conf.hash_query_type,
+                hash_column_name=_fetcher_conf.hash_column_name,
+                config_id=_config_id,
+                filter_keys=_stream_conf.composite_keys,
+                filter_values=payload.composite_keys,
+                hours=10240,
```

Comment:
Any specific reason for this default that we should be aware of?

Reply:
I missed this, updated to config based.